JoinColony/colonyCDapp
# [Notifications] feat: display new colony version notifications

## Description

This PR handles notifications when a new colony version is added to the network.
[Block ingestor PR](https://github.com/JoinColony/block-ingestor/pull/276/)

## Testing

Actually pretty straightforward thanks to @area fixing the scripts :pray: 

1. Open up `planex` as `leela` and give `amy` Owner permissions.
![image](https://github.com/user-attachments/assets/2c0d190d-5e8e-4cca-be13-46a7282e4033)
2. Open up `planex` as `amy` just for her to get the notification data set up
3. In a terminal run `npm run hardhat`
![image](https://github.com/user-attachments/assets/8877373e-6d63-43c9-86ce-3fb2fca3155c)
4. Run the following lines by line (it will take some time):
```
deployNext=require("./scripts/deployOldUpgradeableVersion.js").deployNextColonyVersion
cn = await artifacts.require("IColonyNetwork").at("0x777760996135F0791E2e1a74aFAa060711197777")
await deployNext(cn)
```
![image](https://github.com/user-attachments/assets/1a1b5ab4-d692-4c16-9878-3f446d7bd9db)
5. Verify that as `leela` you have received a notification for all the colonies you have the `Owner` role in, while `amy` should get it just for `planex`
![image](https://github.com/user-attachments/assets/40a3e7e1-7d6e-4349-a2b9-696438fecd63)
6. Verify that clicking on the notification takes you on the respected colony's advanced page


## Diffs

**New stuff** ✨

* `NewColonyVersion` notification
* block ingestor now sends notifications for it and added a helper called `getAllColoniesWithRootPermissionHolders` there

Resolves #3193
